probe-rs/src/flashing/flasher.rs

```diff
@@ -433,17 +464,30 @@ impl Flasher {
 
         let result = if self.flash_algorithm.pc_read.is_some() {
             self.run_verify(session, progress, |active, data| {
-                fill_pages(data, progress, |address, data| {
-                    active.read_flash(address, data)
+                Box::pin(async move {
+                    fill_pages(active, data, progress, |active, address, data| {
+                        Box::pin(async move { active.read_flash(address, data).await })
+                    })
+                    .await
                 })
             })
+            .await
         } else {
+            // TODO: Currently we cannot capture outside names as we want because of typesystem
```

Comment:
We have the very suboptimal problem that we can explicitely tie the lifetime of the future returned from the closure to an input argument. If we capture a reference in the closure, there is no way of naming that lifetime and making the condition that it lives longer than the future. This will probably not happen either. So I opted for just using the active flasher to access the core. This so we can use the same fill_pages function as above.
Of course that means that we can move the `if self.flash_algorithm.pc_read.is_some()` branch inwards. I am not sure I like this, as it has a bit of an overhead, but everything else requires more boilerplate code to pass different closure args.

What to do?

Reply:
This actually introduces a bug, you can't always read memory when the flash algorithm is initialized. The `else` branch shouldn't run its code in `run_verify`

Reply:
Good point! Forgot about that one.